### PR TITLE
feat: Support instantiating `GoogleAuth` with an `AuthClient`

### PR DESCRIPTION
**Pull request by danielbankhead**
This will allow downstream clients to support features like Downscoped Tokens without breaking changes

### CODE REVIEW

File: samples/downscopedclient.js [hunk 1]
```diff
@@ -75,31 +77,15 @@ async function main() {
...
-      },
-      authorizeRequest: async opts => {
-        opts = opts || {};
-        const url = opts.url || opts.uri;
-        const headers = await oauth2Client.getRequestHeaders(url);
-        opts.headers = Object.assign(opts.headers || {}, headers);
-        return opts;
-      },
-    },
+    authClient: new GoogleAuth({auth: oauth2Client}),
```
Comment (bcoe) [inline samples/downscopedclient.js:80]: Well, this is much nicer.

File: samples/test/downscoping-with-cab.test.js [hunk 2]
```diff
@@ -30,9 +30,9 @@ const {promisify} = require('util');
 const exec = promisify(cp.exec);
 // Copy values from the output of samples/scripts/downscoping-with-cab-setup.js.
 // GCS bucket name.
-const bucketName = 'cab-int-bucket-z2zsauf4sj';
+const bucketName = 'cab-int-bucket-brd3qlsuok';
 // GCS object name.
-const objectName = 'cab-first-z2zsauf4sj.txt';
+const objectName = 'cab-first-"brd3qlsuok.txt';
```
Comment (bcoe) [inline samples/test/downscoping-with-cab.test.js:35]: curious why we changed this object name.

**Review by bcoe — APPROVED** [review]
This implementation seems super clean, great work.

I'm always somewhat cautious with updates to the client library, perhaps you could link and run the storage sample/system tests against this branch? at which point let's ship.

File: samples/test/downscoping-with-cab.test.js (hunk 2 above)
Comment (danielbankhead) [reply samples/test/downscoping-with-cab.test.js:35]: Good question: The bucket and object previously didn’t exist - and the test has silently failed until this PR as we were asserting a string that existed in the error message (‘first’).

**Pull request merged**